pkg/postage/batchstore/store.go

```diff
@@ -5,62 +5,139 @@
 package batchstore
 
 import (
+	"errors"
 	"math/big"
 
 	"github.com/ethersphere/bee/pkg/postage"
 	"github.com/ethersphere/bee/pkg/storage"
 )
 
 const (
-	batchKeyPrefix = "batchKeyPrefix"
-	stateKey       = "stateKey"
+	batchKeyPrefix  = "batch"
+	valueKeyPrefix  = "value"
+	chainStateKey   = "chain"
+	reserveStateKey = "reserve"
 )
 
+// store implements postage.Storer
 type store struct {
-	store storage.StateStorer // State store backend to persist batches.
+	store         storage.StateStorer                      // State store backend to persist batches.
+	cs            *postage.ChainState                      // the chain state
+	rs            *reserveState                            // the reserve state
+	unreserveFunc func(batchID []byte, radius uint8) error // unreserve function
 }
 
 // New constructs a new postage batch store.
-func New(st storage.StateStorer) postage.Storer {
-	return &store{st}
+// It initialises both chain state and reserve state from the persistent state store
+func New(st storage.StateStorer, unreserveFunc func(batchID []byte, radius uint8) error) (postage.Storer, error) {
+	cs := &postage.ChainState{}
+	err := st.Get(chainStateKey, cs)
+	if err != nil {
+		if !errors.Is(err, storage.ErrNotFound) {
+			return nil, err
+		}
+		cs = &postage.ChainState{
+			Block: 0,
+			Total: big.NewInt(0),
+			Price: big.NewInt(0),
+		}
+	}
+	rs := &reserveState{}
+	err = st.Get(chainStateKey, rs)
```

Comment:
wrong key

Reply:
wow  good  spotting, clearly we need a test for persistance of resere  state